src/PrestaShopBundle/Resources/config/routing/admin/sell/customer_service/merchandise_return.yml

```diff
@@ -23,3 +23,66 @@ admin_merchandise_returns_filter:
     redirectRoute: admin_merchandise_returns_index
     _legacy_controller: AdminReturn
 #    _legacy_link: AdminReturn:submitFilterorder_return
+
+admin_merchandise_returns_products_filter:
+  path: /{merchandiseReturnId}/edit
+  methods: [POST]
+  defaults:
+    _controller: PrestaShopBundle:Admin\Common:searchGrid
+    gridDefinitionFactoryServiceId: prestashop.core.grid.definition.factory.merchandise_return_products
+    redirectRoute: admin_merchandise_returns_edit
+    redirectQueryParamsToKeep:
+      - 'merchandiseReturnId'
+    _legacy_controller: AdminReturn
+#    _legacy_link: AdminReturn:submitFilterorder_return
+  requirements:
+    merchandiseReturnId: \d+
+
+admin_merchandise_returns_edit:
+  path: /{merchandiseReturnId}/edit
+  methods: [GET]
+  defaults:
+    _controller: PrestaShopBundle:Admin/Sell/CustomerService/MerchandiseReturn:edit
+    _legacy_controller: AdminReturn
+#    _legacy_link: AdminReturn:updateorder_return
+#    _legacy_parameters:
+#      id_order_return: merchandiseReturnId
+  requirements:
+    merchandiseReturnId: \d+
+
+# @todo wonder If I can have the same action in controller for this route or do I need to create another one to clearly separate them?
```

Comment:
Since you use the same controller anyway I think you can combine them and simply have `admin_merchandise_returns_edit`
This is usually what we do, although your suggestion is also very valid